exporters/metric/cortex/cortex.go

```diff
@@ -157,28 +181,34 @@ func (e *Exporter) ConvertToTimeSeries(checkpointSet export.CheckpointSet) ([]*p
 	return timeSeries, nil
 }
 
+// createTimeSeries is a helper function to create a timeseries from a value and labels
+func createTimeSeries(record metric.Record, value apimetric.Number, extraLabels ...string) *prompb.TimeSeries {
+	sample := prompb.Sample{
+		Value:     value.CoerceToFloat64(record.Descriptor().NumberKind()),
+		Timestamp: record.EndTime().Unix(),
+	}
+
+	labels := createLabelSet(record, extraLabels...)
+
+	return &prompb.TimeSeries{
+		Samples: []prompb.Sample{sample},
+		Labels:  labels,
+	}
+}
+
 // convertFromSum returns a single TimeSeries based on a Record with a Sum aggregation
 func convertFromSum(record metric.Record, sum aggregation.Sum) (*prompb.TimeSeries, error) {
 	// Get Sum value
 	value, err := sum.Sum()
 	if err != nil {
 		return nil, err
 	}
-	// Create sample from Sum value
-	sample := prompb.Sample{
-		Value:     value.CoerceToFloat64(record.Descriptor().NumberKind()),
-		Timestamp: record.EndTime().Unix(), // Convert time to Unix (int64)
-	}
 
-	// Create labels, including metric name
+	// Create TimeSeries
 	name := sanitize(record.Descriptor().Name())
-	labels := createLabelSet(record, "__name__", name)
-
-	// Create TimeSeries and return
-	tSeries := &prompb.TimeSeries{
-		Samples: []prompb.Sample{sample},
-		Labels:  labels,
-	}
+	// Note: Cortex requires the name label to be in the format "__name__".
```

Comment:
This duplicated pattern indicates that a refactor might be beneficial here. If `createLabelSet` accepted `label.KeyValue` pairs instead of runtime-guaranteed string pairs it would move the pair checking part of that code to be performed by the compiler and allow this to be templated with a common `label.Key`.

Reply:
To clarify, instead of strings, `createLabelSet` would take in `...label.KeyValue`? How would you recommend merging the three sets of labels (record labels, resource labels, and extra labels)? The desired behavior is to give precedence to labels the exporter adds (like `"__name__"`, `"le"`, and `"quantile"`), then record labels, then resource labels.

Reply:
> To clarify, instead of strings, `createLabelSet` would take in `...label.KeyValue`?

Correct.

> How would you recommend merging the three sets of labels (record labels, resource labels, and extra labels)? The desired behavior is to give precedence to labels the exporter adds (like `"__name__"`, `"le"`, and `"quantile"`), then record labels, then resource labels.

Similar to what you are already doing in `createLabelSet` would be my suggestion. You would iterate through the labels in the order of precedence and overwrite any duplicate `Key` appropriately. So, instead of looping through the extra label strings last you would just loop through the extra label `label.KeyValue` last.

This can also be done in a separate PR. I didn't make that clear in my original comment, sorry. If we want to omit it from this PR we can create an issue to track it.

Reply:
There's an example of merging that leaves duplicate keys in place in the dogstatsd exporter, see `func (e *Exporter) AppendTags(rec export.Record, res *resource.Resource, buf *bytes.Buffer)`.

The `api/label` package has a MergeIterator to support merging label sets, which is another option, which is used to give precedence to labels over resources, it could be used here as well.

Reply:
That makes sense and would be a cleaner API. I'll file this in another PR.